backends/bmv2/simple_switch/simpleSwitch.cpp

```diff
@@ -895,6 +895,10 @@ SimpleSwitchBackend::convert(const IR::ToplevelBlock* tlb) {
         evaluator,
         new VisitFunctor([this, evaluator]() { toplevel = evaluator->getToplevelBlock(); }),
     };
+
+    auto hook = options.getDebugHook();
```

Comment:
was that just for debugging?

Reply:
Mostly; it prints the passes as they get executed, but it also enables you to dump the program after each pass.